torchgeo/datasets/loveda.py

```diff
@@ -0,0 +1,261 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""LoveDA dataset."""
+
+import glob
+import os
+from typing import Callable, Dict, List, Optional
+
+import numpy as np
+import torch
+from PIL import Image
+from torch import Tensor
+
+from .geo import VisionDataset
+from .utils import download_and_extract_archive
+
+
+class LoveDA(VisionDataset):
+    """LoveDA dataset
+
+    The LoveDA <https://github.com/Junjue-Wang/LoveDA> datataset is a semantic segmentation
+    dataset.
+
+    Dataset features:
+
+    * 2713 urban scene and 3274 rural scene HSR images with a total of 166768 annotated objects
+    from Nanjing, Changzhou and Wuhan cities
+    * dataset comes with predefined train, validation, and test set
+
+    Dataset format:
+
+    * images are three-channel pngs with dimension 1024x1024
+    * segmentation masks are single-channel pngs
+
+    Dataset classes:
+
+    1. background
+    2. building
+    3. road
+    4. water
+    5. barren
+    6. forestn
+    7. agriculture
+
+    No-data regions assigned with 0 and should be ignored.
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * <https://openreview.net/forum?id=bLBIbVaGDu>
+    """
+
+    # url of train split
+    url = "https://drive.google.com/file/d/1xbnKVN5aRMlpxISXgutzQO0hPT_b4lMi/view?usp=sharing"
+    # md5 of train split
+    md5 = "de2b196043ed9b4af1690b3f9a7d558f"
+    scenes = ["urban", "rural"]
+    splits = ["train", "val", "test"]
+
+    info_dict = {
+        "train": {
+            "url": "https://drive.google.com/file/d/1xbnKVN5aRMlpxISXgutzQO0hPT_b4lMi/view?usp=sharing",
+            "filename": "Train.zip",
+            "md5": "de2b196043ed9b4af1690b3f9a7d558f",
+        },
+        "val": {
+            "url": "https://drive.google.com/file/d/1yTMfeon1Oc4ia9oCX7r5Yq4C39I0oO_b/view?usp=sharing",
+            "filename": "Val.zip",
+            "md5": "84cae2577468ff0b5386758bb386d31d",
+        },
+        "test": {
+            "url": "https://drive.google.com/file/d/1ON7bWat7u9fGV16stAosdmzMIpcydnVC/view?usp=sharing",
+            "filename": "Test.zip",
+            "md5": "a489be0090465e01fb067795d24e6b47",
+        },
+    }
+
+    classes = [
+        "background",
+        "building",
+        "road",
+        "water",
+        "barren",
+        "forest",
+        "agriculture",
+        "no-data",
+    ]
+
+    def __init__(
+        self,
+        root: str = "./data",
```

Comment:
Might be better if root does not have a default value as we can't make assumptions on users' directory organisation.

Reply:
Every other dataset defaults to `'data'`, so this is fine. For most users, they'll simply use `download=True` and not need to worry about setting a `root`. The only change I would make is:
```suggestion
        root: str = "data",
```